src/app/task/shared/services/task.service.ts

```diff
@@ -1,36 +1,33 @@
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { Task } from "../models/task.model";
+import { Observable } from "rxjs";
 
 @Injectable({
   providedIn: 'root'
 })
 export class TaskService {
-  taskList : Task[] = new Array();
+  BASE_URL = 'http://localhost:3000/api/tasks/';
 
-  constructor(private http: HttpClient) {
-    this.http.get('assets/tasks.json').subscribe((data : any) => {
-      this.taskList = data;
-    });
-  }
+  constructor(private http: HttpClient) {}
 
-  getAllTasks() {
-    return this.taskList;
+  getAllTasks(): Observable<any> {
+    return this.http.get(`${this.BASE_URL}get-all-tasks`);
```

Comment:
Please have a look on how RESTful APIs are created.
Should be something like:
| Method | Endpoint     | Comment            |
|--------|--------------|--------------------|
| GET    | `/tasks`     | get all tasks      |
| GET    | `/tasks/:id` | get one task by id |
| POST   | `/tasks`     | create a new task  |
| PUT    | `/tasks/:id` | update task        |
| DELETE | `/tasks/:id` | delete task        |